src/OrchardCore/OrchardCore.FileStorage.AzureBlob/BlobFileStore.cs

```diff
@@ -262,6 +260,19 @@ public async Task<Stream> GetFileStreamAsync(string path)
             return await blob.OpenReadAsync();
         }
 
+        // Reduces the need to call blob.FetchAttributes, and blob.ExistsAsync,
+        // as Azure Storage Library will perform these actions on OpenReadAsync(). 
+        public async Task<Stream> GetFileStreamAsync(IFileStoreEntry fileStoreEntry)
+        {
+            await _verifyContainerTask;
+
+            var blobFile = fileStoreEntry as BlobFile;
+            if (blobFile == null || blobFile.BlobReference == null)
+                throw new FileStoreException("Cannot get file stream because the file does not exist.");
+
+            return await blobFile.BlobReference.OpenReadAsync();
+        }
```

Comment:
This just reduces head requests, bearing in mind that the azure storage library always calls `FetchAttributes` on `BlobReference.OpenReadAsync` to check for existance.